### ISSUE TITLE
Crash in loadout screen when clicking a weapons-locked ship

### ISSUE THREAD

**Issue by Baezon** [scp-fs2open/fs2open.github.com#4335]
Weapons locked *only*, that is. At `missionshipchoice.cpp:2126` the case where a slot is only weapons locked is not handled. I tried fixing this, but I only broke/exposed more places that don't handle a ship being only weapons locked, and not ship locked. It also erroneously grays out weapons locked ships.

**Comment by jg18** [comment]
Looks like this part of the code treats weapons-locked and ship-locked wing slots as none of empty/filled/locked, which is incorrect.

Even if a wing slot is weapons-locked, we still need to track whether the slot is empty or filled, and the code currently doesn't do this.

IMHO, slot status should obey these constraints:
1. Every slot should be exactly one of empty, filled, or locked.
2. A locked slot should never be ship-locked, weapons-locked, or the player.
3. A ship-locked slot should always be filled.

**Comment by Karajorma** [comment]
That last bit of logic is correct. I never expected anyone would ever lock a ship's weapons but not the ship itself cause that seemed pretty strange. But I expected there would be plenty of FREDders who wanted to lock a ship but still allow the player to choose the weapons available on the ship.

**Comment by Baezon** [comment]
Then you should have disallowed it when you implemented it! Alas, you built it, and now someone has come with a use case for it lol

jg18 said he'll post an update soon, but we discussed this a bit on the Discord; weapons-locked will attempt to fill with default weapons for the class, taking from the pool as necessary, and using <none> otherwise. I'll leave to you whether this is a good idea, but it *is* what would be most useful for me.

**Comment by jg18** [comment]
After more Discord discussion, we've decided to keep weapons-locked independent of ship-locked, even if it wasn't the intended behavior.

Therefore, we're effectively going with the slot constraints I proposed above, which I've updated slightly for clarity.

To enforce these constraints in the code, I propose adding

```
enum class SlotStatus {
	Empty,
	Filled,
	Ship_Locked,
	Locked
};
```
and then replacing `ss_slot_info`'s `int status` with two fields, `SlotStatus status` and `bool weapons_locked`.

 I'd also add a new internal variable
```
ss_slot_info	*Ss_player_slot = nullptr;
```
and then update `Ss_player_slot` as each mission loads/unloads.

While it'll entail a number of code changes, it'll force us to think through what should happen under which circumstances.

Thoughts @karajorma @Baezon @notimaginative @johnafernandez?

**Comment by jg18** [comment]
To clarify, by "a number of code changes", I just mean we'd have to replace all of the `WING_SLOT_*` stuff if we went this route.

**Comment by jg18** [comment]
I can also try for a less invasive PR that doesn't replace `int ss_slot_info::status`.

**Comment by jg18** [comment]
Fixed with PR #4399 which @Baezon verified.

I guess resolving this required starting with the complicated solution to find the simple one. 🙂
